### PR TITLE
refactor: improved placeholders for error messages

### LINKED ISSUE #1271: Convert regex capturing groups in cdkErrorMapper to Named capturing groups
### Environment information

```plain text
N/A
```

### Description

To better manage the exploding list of error translations and capture groups, it'd be better to change all the capturing groups to named groups so we can more confidently do the replacements (error translations)

See https://github.com/aws-amplify/amplify-backend/pull/1270#discussion_r1556376299

### PR DESCRIPTION
**Pull request by fossamagna**
## Problem

**Issue number, if available:**

closes #1271 

## Changes

use [named capturing groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) and replace all {captureGroupName} placeholders with the matching value.

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by Amplifiyer — APPROVED** [review]

File: .changeset/gold-tips-film.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (Amplifiyer) [inline .changeset/gold-tips-film.md:2]:
```suggestion
---
'@aws-amplify/backend-deployer': patch
---

improve placeholders for cdk error messages
```

**Review by Amplifiyer — CHANGES_REQUESTED** [review]

**Review by awsluja — APPROVED** [review]

**Review by awsluja — CHANGES_REQUESTED** [review]
Looks good , just need to add the changeset suggested by @Amplifiyer

**Comment by fossamagna** [comment]
@Amplifiyer @awsluja
Thank you for your review and suggestion.

**Comment by Amplifiyer** [comment]
Thanks @fossamagna, please merge from main to make the github checks happy. We can then merge this.

**Comment by fossamagna** [comment]
@Amplifiyer I merged from main. Thanks.

**Review by Amplifiyer — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**